### ISSUE TITLE
The Dec. 14 trunk snapshots onwards crash when compiling swift-collections trunk for linux

### ISSUE THREAD

**Issue by buttaface** [apple/swift#62671]
**Description**
I get the following extensive error output when simply trying to compile swift-collections trunk, which isn't reproducible with the last 1.0.4 tag of collections:
```
Building for debugging...
Broken instruction!
  return %64 : $()                                // id: %65
in scope
{
/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift:109:19
   parent: {
  /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift:107:17
     parent: @$s17SortedCollections6_BTreeV5IndexV8readNodeyqd__qd__AA01_F0V12UnsafeHandleVyxq__GKXEKlF
    }

  }
Previous, non-contiguous scope set by  dealloc_stack %47 : $@noescape @callee_guaranteed (@guaranteed _Node<Key, Value>.Storage) -> (@out R, @error any Error) // id: %62
in scope
{
/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift:107:17
   parent: @$s17SortedCollections6_BTreeV5IndexV8readNodeyqd__qd__AA01_F0V12UnsafeHandleVyxq__GKXEKlF
  }
Please report a bug on bugs.swift.org
Pass -Xllvm -verify-di-holes=false to disable the verification
SIL verification failed: Basic block contains a non-contiguous lexical scope at -Onone: DS == LastSeenScope
Verifying instruction:
     %64 = tuple (), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":112:3, scope 0 // user: %65
->   return %64 : $(), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:5, scope 0 // id: %65
In function:
sil_scope 1 { loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":107:17 parent @$s17SortedCollections6_BTreeV5IndexV8readNodeyqd__qd__AA01_F0V12UnsafeHandleVyxq__GKXEKlF : $@convention(method) <Ï„_0_0, Ï„_0_1 where Ï„_0_0 : Comparable><Ï„_1_0> (@noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1, Ï„_0_2 where Ï„_0_0 : Comparable> (_Node<Ï„_0_0, Ï„_0_1>.UnsafeHandle) -> (@out Ï„_0_2, @error any Error) for <Ï„_0_0, Ï„_0_1, Ï„_1_0>, @in_guaranteed _BTree<Ï„_0_0, Ï„_0_1>.Index) -> (@out Ï„_1_0, @error any Error) }
sil_scope 2 { loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":109:19 parent 1 }

// _BTree.Index.readNode<A>(_:)
sil [serialized] [always_inline] [ossa] @$s17SortedCollections6_BTreeV5IndexV8readNodeyqd__qd__AA01_F0V12UnsafeHandleVyxq__GKXEKlF : $@convention(method) <Key, Value where Key : Comparable><R> (@noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1, Ï„_0_2 where Ï„_0_0 : Comparable> (_Node<Ï„_0_0, Ï„_0_1>.UnsafeHandle) -> (@out Ï„_0_2, @error any Error) for <Key, Value, R>, @in_guaranteed _BTree<Key, Value>.Index) -> (@out R, @error any Error) {
// %0 "$return_value"                             // user: %52
// %1 "body"                                      // users: %47, %3
// %2 "self"                                      // users: %44, %7, %4
bb0(%0 : $*R, %1 : $@noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1, Ï„_0_2 where Ï„_0_0 : Comparable> (_Node<Ï„_0_0, Ï„_0_1>.UnsafeHandle) -> (@out Ï„_0_2, @error any Error) for <Key, Value, R>, %2 : $*_BTree<Key, Value>.Index):
  debug_value %1 : $@noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1, Ï„_0_2 where Ï„_0_0 : Comparable> (_Node<Ï„_0_0, Ï„_0_1>.UnsafeHandle) -> (@out Ï„_0_2, @error any Error) for <Key, Value, R>, let, name "body", argno 1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":108:7, scope 1 // id: %3
  debug_value %2 : $*_BTree<Key, Value>.Index, let, name "self", argno 2, implicit, expr op_deref, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":107:17, scope 1 // id: %4
  debug_value undef : $any Error, var, name "$error", argno 3, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":109:5, scope 1 // id: %5
  %6 = alloc_stack [lexical] $_BTree<Key, Value>.Index, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:12, scope 2 // users: %19, %69, %63, %43, %7
  copy_addr %2 to [init] %6 : $*_BTree<Key, Value>.Index, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:12, scope 2 // id: %7
  %8 = string_literal utf8 "SortedCollections/_BTree.Index.swift", loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // user: %10
  %9 = integer_literal $Builtin.Word, 36, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // user: %14
  %10 = builtin "ptrtoint_Word"(%8 : $Builtin.RawPointer) : $Builtin.Word, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // user: %14
  br bb1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // id: %11

bb1:                                              // Preds: bb0
  %12 = integer_literal $Builtin.Int8, 2, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // user: %14
  br bb2, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // id: %13

bb2:                                              // Preds: bb1
  %14 = struct $StaticString (%10 : $Builtin.Word, %9 : $Builtin.Word, %12 : $Builtin.Int8), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // user: %57
  %15 = integer_literal $Builtin.Int64, 110, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // user: %16
  %16 = struct $UInt (%15 : $Builtin.Int64), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:11, scope 2 // user: %57
  br bb3, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // id: %17

bb3:                                              // Preds: bb2
  %18 = metatype $@thin Int.Type, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2
  %19 = struct_element_addr %6 : $*_BTree<Key, Value>.Index, #_BTree.Index.slot, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %20
  %20 = load [trivial] %19 : $*Int, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %21
  %21 = struct_extract %20 : $Int, #Int._value, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %23
  %22 = integer_literal $Builtin.Int64, -1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %23
  %23 = builtin "cmp_eq_Int64"(%21 : $Builtin.Int64, %22 : $Builtin.Int64) : $Builtin.Int1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %25
  %24 = integer_literal $Builtin.Int1, -1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %25
  %25 = builtin "xor_Int1"(%23 : $Builtin.Int1, %24 : $Builtin.Int1) : $Builtin.Int1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %27
  %26 = integer_literal $Builtin.Int1, -1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %27
  %27 = builtin "int_expect_Int1"(%25 : $Builtin.Int1, %26 : $Builtin.Int1) : $Builtin.Int1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %28
  cond_br %27, bb4, bb5, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // id: %28

bb4:                                              // Preds: bb3
  br bb6, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // id: %29

bb5:                                              // Preds: bb3
  %30 = string_literal utf8 "Assertion failed", loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %32
  %31 = integer_literal $Builtin.Word, 16, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %34
  %32 = builtin "ptrtoint_Word"(%30 : $Builtin.RawPointer) : $Builtin.Word, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %34
  %33 = integer_literal $Builtin.Int8, 2, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %34
  %34 = struct $StaticString (%32 : $Builtin.Word, %31 : $Builtin.Word, %33 : $Builtin.Int8), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %57
  %35 = string_literal utf8 "Invalid operation to read end index.", loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %40
  %36 = integer_literal $Builtin.Word, 36, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %40
  %37 = integer_literal $Builtin.Int1, -1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %40
  %38 = metatype $@thin String.Type, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %40
  // function_ref String.init(_builtinStringLiteral:utf8CodeUnitCount:isASCII:)
  %39 = function_ref @$sSS21_builtinStringLiteral17utf8CodeUnitCount7isASCIISSBp_BwBi1_tcfC : $@convention(method) (Builtin.RawPointer, Builtin.Word, Builtin.Int1, @thin String.Type) -> @owned String, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %40
  %40 = apply %39(%35, %36, %37, %38) : $@convention(method) (Builtin.RawPointer, Builtin.Word, Builtin.Int1, @thin String.Type) -> @owned String, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %57
  br bb7, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // id: %41

bb6:                                              // Preds: bb4
  %42 = tuple (), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2
  destroy_addr %6 : $*_BTree<Key, Value>.Index, loc "<compiler-generated>":0:0, scope 2 // id: %43
  %44 = struct_element_addr %2 : $*_BTree<Key, Value>.Index, #_BTree.Index.node, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:21, scope 2 // user: %45
  %45 = load [trivial] %44 : $*Unmanaged<_Node<Key, Value>.Storage>, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:21, scope 2 // user: %49
  // function_ref closure #1 in _BTree.Index.readNode<A>(_:)
  %46 = function_ref @$s17SortedCollections6_BTreeV5IndexV8readNodeyqd__qd__AA01_F0V12UnsafeHandleVyxq__GKXEKlFqd__AH7StorageCyxq__GKXEfU_ : $@convention(thin) <Ï„_0_0, Ï„_0_1 where Ï„_0_0 : Comparable><Ï„_1_0> (@guaranteed _Node<Ï„_0_0, Ï„_0_1>.Storage, @noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1, Ï„_0_2 where Ï„_0_0 : Comparable> (_Node<Ï„_0_0, Ï„_0_1>.UnsafeHandle) -> (@out Ï„_0_2, @error any Error) for <Ï„_0_0, Ï„_0_1, Ï„_1_0>) -> (@out Ï„_1_0, @error any Error), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:51, scope 2 // user: %47
  %47 = partial_apply [callee_guaranteed] [on_stack] %46<Key, Value, R>(%1) : $@convention(thin) <Ï„_0_0, Ï„_0_1 where Ï„_0_0 : Comparable><Ï„_1_0> (@guaranteed _Node<Ï„_0_0, Ï„_0_1>.Storage, @noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1, Ï„_0_2 where Ï„_0_0 : Comparable> (_Node<Ï„_0_0, Ï„_0_1>.UnsafeHandle) -> (@out Ï„_0_2, @error any Error) for <Ï„_0_0, Ï„_0_1, Ï„_1_0>) -> (@out Ï„_1_0, @error any Error), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:51, scope 2 // users: %68, %62, %48
  %48 = convert_function %47 : $@noescape @callee_guaranteed (@guaranteed _Node<Key, Value>.Storage) -> (@out R, @error any Error) to $@noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1 where Ï„_0_0 : AnyObject> (@guaranteed Ï„_0_0) -> (@out Ï„_0_1, @error any Error) for <_Node<Key, Value>.Storage, R>, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:51, scope 2 // user: %52
  %49 = struct_extract %45 : $Unmanaged<_Node<Key, Value>.Storage>, #Unmanaged._value, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 2 // user: %50
  %50 = unmanaged_to_ref %49 : $@sil_unmanaged _Node<Key, Value>.Storage to $_Node<Key, Value>.Storage, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 2 // user: %51
  %51 = unchecked_ownership_conversion %50 : $_Node<Key, Value>.Storage, @unowned to @guaranteed, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 2 // users: %52, %67, %60
  try_apply %48(%0, %51) : $@noescape @callee_guaranteed @substituted <Ï„_0_0, Ï„_0_1 where Ï„_0_0 : AnyObject> (@guaranteed Ï„_0_0) -> (@out Ï„_0_1, @error any Error) for <_Node<Key, Value>.Storage, R>, normal bb9, error bb10, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 2 // id: %52

bb7:                                              // Preds: bb5
  %53 = integer_literal $Builtin.Int32, 1, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %54
  %54 = struct $UInt32 (%53 : $Builtin.Int32), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %57
  br bb8, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // id: %55

bb8:                                              // Preds: bb7
  // function_ref _assertionFailure(_:_:file:line:flags:)
  %56 = function_ref @$ss17_assertionFailure__4file4line5flagss5NeverOs12StaticStringV_SSAHSus6UInt32VtF : $@convention(thin) (StaticString, @guaranteed String, StaticString, UInt, UInt32) -> Never, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2 // user: %57
  %57 = apply %56(%34, %40, %14, %16, %54) : $@convention(thin) (StaticString, @guaranteed String, StaticString, UInt, UInt32) -> Never, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":110:5, scope 2
  unreachable , scope 2                           // id: %58

bb9(%59 : $()):                                   // Preds: bb6
  end_borrow %51 : $_Node<Key, Value>.Storage, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 2 // id: %60
  %61 = tuple (), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 2
  dealloc_stack %47 : $@noescape @callee_guaranteed (@guaranteed _Node<Key, Value>.Storage) -> (@out R, @error any Error), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":107:17, scope 1 // id: %62
  dealloc_stack %6 : $*_BTree<Key, Value>.Index, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":112:3, scope 1 // id: %63
  %64 = tuple (), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":112:3, scope 2 // user: %65
  return %64 : $(), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:5, scope 2 // id: %65

// %66                                            // user: %70
bb10(%66 : @owned $any Error):                    // Preds: bb6
  end_borrow %51 : $_Node<Key, Value>.Storage, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 2 // id: %67
  dealloc_stack %47 : $@noescape @callee_guaranteed (@guaranteed _Node<Key, Value>.Storage) -> (@out R, @error any Error), loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:26, scope 1 // id: %68
  dealloc_stack %6 : $*_BTree<Key, Value>.Index, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:71, scope 1 // id: %69
  throw %66 : $any Error, loc "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift":111:71, scope 2 // id: %70
} // end sil function '$s17SortedCollections6_BTreeV5IndexV8readNodeyqd__qd__AA01_F0V12UnsafeHandleVyxq__GKXEKlF'

Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the crash backtrace.
Stack dump:
0.      Program arguments: /home/butta/swift-DEVELOPMENT-SNAPSHOT-2022-12-14-a-ubuntu20.04/usr/bin/swift-frontend -frontend -emit-module -experimental-skip-non-inlinable-function-bodies-without-types /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+BidirectionalCollection.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+CustomDebugStringConvertible.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+CustomReflectable.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+Invariants.swift "/home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+Partial RangeReplaceableCollection.swift" /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+Sequence.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+SubSequence.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree+UnsafeCursor.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Builder.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_FixedSizeArray.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node+CustomDebugString.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node+Testing.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node.Splinter.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node.Storage.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node.UnsafeHandle+CustomDebugStringConvertible.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node.UnsafeHandle+Deletion.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node.UnsafeHandle+Insertion.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node.UnsafeHandle.swift /home/butta/swift-collections/Sources/SortedCollections/BTree/_Node.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+BidirectionalCollection.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Codable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+CustomReflectable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+CustomStringConvertible.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Equatable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+ExpressibleByDictionaryLiteral.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Hashable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Initializers.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Keys.swift "/home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Partial RangeReplaceableCollection.swift" /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Sendable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Sequence.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+SubSequence.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Subscripts.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary+Values.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary.Index.swift /home/butta/swift-collections/Sources/SortedCollections/SortedDictionary/SortedDictionary.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+BidirectionalCollection.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Codable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+CustomReflectable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+CustomStringConvertible.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Equatable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+ExpressibleByArrayLiteral.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Hashable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Initializers.swift "/home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Partial RangeReplaceableCollection.swift" /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Sendable.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Sequence.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+SetAlgebra.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+SubSequence.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet+Subscripts.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet.Index.swift /home/butta/swift-collections/Sources/SortedCollections/SortedSet/SortedSet.swift /home/butta/swift-collections/Sources/SortedCollections/Utilities/Assertions.swift -target x86_64-unknown-linux-gnu -disable-objc-interop -I /home/butta/swift-collections/.build/x86_64-unknown-linux-gnu/debug -color-diagnostics -enable-testing -g -module-cache-path /home/butta/swift-collections/.build/x86_64-unknown-linux-gnu/debug/ModuleCache -swift-version 5 -Onone -D SWIFT_PACKAGE -D DEBUG -new-driver-path /home/butta/swift-DEVELOPMENT-SNAPSHOT-2022-12-14-a-ubuntu20.04/usr/bin/swift-driver -empty-abi-descriptor -resource-dir /home/butta/swift-DEVELOPMENT-SNAPSHOT-2022-12-14-a-ubuntu20.04/usr/lib/swift -enable-anonymous-context-mangled-names -module-name SortedCollections -emit-module-doc-path /home/butta/swift-collections/.build/x86_64-unknown-linux-gnu/debug/SortedCollections.swiftdoc -emit-module-source-info-path /home/butta/swift-collections/.build/x86_64-unknown-linux-gnu/debug/SortedCollections.swiftsourceinfo -emit-dependencies-path /home/butta/swift-collections/.build/x86_64-unknown-linux-gnu/debug/SortedCollections.build/SortedCollections.emit-module.d -parse-as-library -o /home/butta/swift-collections/.build/x86_64-unknown-linux-gnu/debug/SortedCollections.swiftmodule
1.      Swift version 5.8-dev (LLVM 3552b731b61db92, Swift 9cebd696545a8f4)
2.      Compiling with the current language version
3.      While verifying SIL function "@$s17SortedCollections6_BTreeV5IndexV8readNodeyqd__qd__AA01_F0V12UnsafeHandleVyxq__GKXEKlF".
 for 'readNode(_:)' (at /home/butta/swift-collections/Sources/SortedCollections/BTree/_BTree.Index.swift:107:12)
```
I first [saw this on my Android CI a couple days ago](https://github.com/buttaface/swift-android-sdk/actions/runs/3719338426/jobs/6308141533#step:19:124).

**Steps to reproduce**
`swift build`

**Expected behavior**
This wasn't crashing with the Dec. 13 trunk snapshot and before, so this appears to be a recent trunk regression.

**Environment**
- Swift 5.8 trunk snapshot for Ubuntu 20.04 x86_64 from Dec.14 and later, building the latest collections commit 53a8adc54

**Comment by buttaface** [comment]
Still seeing this with the latest Jan. 8 trunk snapshot for Ubuntu 20.04 x86_64, though I've since noticed that it goes away when building for release with `-c release`.

@lorentey, do you think you can get this looked into?

**Comment by buttaface** [comment]
Still hitting this with the latest Jan. 23 trunk snapshot for Ubuntu 20.04 x86_64, @nate-chandler, any idea?

**Comment by nate-chandler** [comment]
CC: @adrian-prantl

**Comment by buttaface** [comment]
I just checked and the latest official snapshots for Ubuntu 20.04 x86_64, trunk from Feb. 13 and the 5.8 branch from Feb. 9, still crash with this same error. Since 5.8 presumably will be released next month, this should be fixed before that.

@adrian-prantl, any idea who is best situated to look into this? Can someone confirm if it's crashing on macOS too?

**Comment by adrian-prantl** [comment]
From the looks of it the dealloc_stack was created with the wrong scope. I'll need to see if I can reproduce this. If you can give me a smaller reproducer (Linux or macOS), that would be a big help.

**Comment by QuietMisdreavus** [comment]
@adrian-prantl This seems to be affecting Swift-DocC's test suite, cf. https://github.com/apple/swift-docc/pull/486

**Comment by lorentey** [comment]
> SortedCollections/BTree/_BTree.Index.swift":112:3, scope 0 

This source file is not production ready, and it isn't part of any Swift Collections release. Is someone building this package from its `main` branch?

**Comment by buttaface** [comment]
> Is someone building this package from its main branch?

Yes, as I noted in my OP, I'm building trunk.

**Comment by buttaface** [comment]
@lorentey, is it okay that this main-branch code started crashing the compiler a couple months ago then?

**Comment by adrian-prantl** [comment]
This may or may not have been fixed by https://github.com/apple/swift/pull/63811

**Comment by lorentey** [comment]
@buttaface Compiler crashes aren't okay, ever. However, branch checkouts are (by definition) unstable -- the contents of the `main` branch can (and will) change at will, with no advance notice. The package only promises a level of stability for tagged releases.

**Comment by buttaface** [comment]
@lorentey, alright, that's all well and good, as then I _am_ using these trunk branches for their intended purpose, to smoke out these bugs before they get into a tagged release like the upcoming 5.8 compiler.

I'll check if the pull Adrian linked above fixes the problem in the next trunk snapshot that contains it, but if it doesn't, maybe you can provide him with the minimal reproducer he asked for, since you're much more familar with this code.

**Comment by buttaface** [comment]
@adrian-prantl, I am unable to reproduce with the latest Feb. 23 trunk snapshot, implying Joe's pull did fix the issue. The latest Feb. 23 5.8 snapshot still crashes though, so please try merging it to the 5.8 branch also and let's confirm that was the fix.

**Comment by buttaface** [comment]
The linked pull #63811 was merged into the 5.8 branch also a couple days ago, will confirm that it stops this crash with the next official 5.8 build containing it and close this issue if so.

**Comment by buttaface** [comment]
Confirmed that this was fixed in trunk by # 63811 and 5.8 by #63891, so closing.
